manage

```diff
@@ -445,82 +447,91 @@ $app"
+      # reorder queue list to prioritize app refresh and file update actions
+      queue="$(reorder_list "$queue")"
     fi
-    # reorder queue list to prioritize app refresh and file update actions
-    queue="$(reorder_list "$queue")"
     #echo "length of queue is $(echo "$queue" | wc -l)"
     
-    if [ "${current_line_num}" -le "$(echo "$queue" | wc -l)" ];then #if we have not iterated through the entire queue length
-      #echo "current position in queue is ${current_line_num}"
-      #echo "queue is '$queue'"
-      
-      line="$(echo "$queue" | sed -n "${current_line_num}"p)"
-      #echo "Now handling request: '$line'"
-      
-      #indicate current action in current line of $queue
-      queue="$(echo "$queue" | sed "${current_line_num}s/$/;in-progress/")"
-      
-      #get first word of this line - the action. Subsequent words are the name of the app.
-      action="$(echo "$line" | awk -F';' '{print $1}')"
-      app="$(echo "$line" | awk -F';' '{print $2}')"
-      
-      #Set terminal title
-      echo -ne "\e]0;${action^}ing ${app}\a" | sed 's/Updateing/Updating/g'
-      
-      #refresh the list in queue-viewer window
-      write_list "$queue"
-      
-      #echo "Action is $action, app is $app"
-      
+    #exit loop if queue is complete and no new actions were added to the queue
+    if [ "${current_line_num}" -gt "$(echo "$queue" | wc -l)" ];then
+      break
+    fi
+    
+    #echo "current position in queue is ${current_line_num}"
+    #echo "queue is '$queue'"
+    
+    line="$(echo "$queue" | sed -n "${current_line_num}"p)"
+    #echo "Now handling request: '$line'"
+    
+    #indicate current action in current line of $queue
+    queue="$(echo "$queue" | sed "${current_line_num}s/$/;in-progress/")"
+    
+    #get first word of this line - the action. Subsequent words are the name of the app.
+    action="$(echo "$line" | awk -F';' '{print $1}')"
+    app="$(echo "$line" | awk -F';' '{print $2}')"
+    
+    #Set terminal title
+    echo -ne "\e]0;${action^}ing ${app}\a" | sed 's/Updateing/Updating/g'
+    
+    #refresh the list in queue-viewer window as a background process - skip it if the list is still refreshing from last loop iteration; in game dev this is 'frame skipping'
+    if [ -z "$write_list_pid" ];then
+      write_list "$queue" &
+      write_list_pid=$!
+    elif ! process_exists "$write_list_pid" ;then
+      write_list "$queue" &
+      write_list_pid=$!
+    fi
+    
+    #run manage script for app installs, uninstalls, or updates. Not for file-updates or app refreshes.
+    if [ "$action" == update ] && [[ "$app" =~ ^filelist:.* ]];then #list of files separated by ':'
+      # use the second argument to pass our list of files to update
+      updatable_apps='' updatable_files="$(echo "$app" | sed 's/^filelist://g' | tr ':' '\n')" no_status=true update_now_cli
+      exitcode=$?
+    else
+      echo manage
       "${DIRECTORY}/manage" "$action" "$app"
+      echo done
```

Comment:
looks to me like some debugging output is still enabled here

Reply:
I noticed and fixed this already.